Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift

```diff
@@ -25,6 +25,19 @@ public class StarknetProvider: StarknetProviderProtocol {
         self.init(starknetChainId: starknetChainId, url: url)
     }
 
+    public init(starknetChainId: StarknetChainId, url: URL, urlSession: URLSession) {
```

Comment:
I also considered the solution with default parameter, but I think it's better if the default case (no URL session) is handled by `HttpNetworkProvider` itself